### PR TITLE
Fix det_exact for ruby object

### LINKED ISSUE #496: NMatrix#det_exact not working on :object dtype
```
t = N[[2,2],[3,4], dtype: :object]
t.det # Gives NotImplementedError, more or less as expected
t.det_exact
=> 
Cannot create ruby object
DataTypeError: Conversion to RubyObject requested from unknown/invalid data type (did you try to convert from a VALUE?)
```

I believe the problem is somewhere around https://github.com/SciRuby/nmatrix/blob/0de5c43983658cc27b30d712048e931adc9ec8e8/ext/nmatrix/math.cpp#L197

I recall seeing this type of error when an operator wasn't properly implemented on the `RubyObject` class, but I could be mistaken about that.

This issue blocks https://github.com/symengine/symengine.rb/issues/36.

### PR DESCRIPTION
**Pull request by isuruf**
Fixes #496.

### CODE REVIEW

File: spec/math_spec.rb [hunk 1]
```diff
@@ -968,13 +967,19 @@
                 end
         end
         it "computes the determinant of 2x2 matrix" do
-          expect(@a.det).to be_within(@err).of(-2)
+          if dtype != :object
```
Comment (mohawkjohn) [inline spec/math_spec.rb]: It seems like you're not testing `#det` with Ruby objects anymore.

Comment (isuruf) [inline spec/math_spec.rb]: I'm not testing `#det` only `#det_exact`. Both were disabled in line 949 before

Comment (mohawkjohn) [inline spec/math_spec.rb]: Ahh! I see now.

File: ext/nmatrix/ruby_nmatrix.c [hunk 2]
```diff
@@ -3006,12 +3006,12 @@ static VALUE nm_det_exact(VALUE self) {
   nm::dtype_t dtype = NM_DTYPE(self);
   nm_math_det_exact(NM_SHAPE0(self), NM_STORAGE_DENSE(self)->elements, NM_SHAPE0(self), NM_DTYPE(self), result);
 
+  VALUE to_return;
   if (dtype == nm::RUBYOBJ) {
```
Comment (mohawkjohn) [inline ext/nmatrix/ruby_nmatrix.c]: Good catch on this. @cjfuller Just want to check this with you since you wrote the register/unregister functions.

**Pull request merged**